deepforest/evaluate.py

```diff
@@ -164,3 +164,115 @@ def evaluate(predictions, ground_df, root_dir, iou_threshold=0.4, savedir=None):
+    Args:
+        df: a pandas dataframe with columns name, xmin, xmax, ymin, ymax, label. The 'name' column should be the path relative to the location of the file.
+        summarize: Whether to group statistics by plot and overall score
+        image_coordinates: Whether the current boxes are in coordinate system of the image, e.g. origin (0,0) upper left.
+        root_dir: Where to search for image names in df, only needed if savedir is supplied
+        savedir: optional directory to save image with overlaid predictions and annotations
+    Returns:
+        result: pandas dataframe with crown ids of prediciton and ground truth and the IoU score.
+    """
+    plot_names = predictions["image_path"].unique()
+    if len(plot_names) > 1:
+        raise ValueError("More than one plot passed to image crown: {}".format(plot_name))
+    else:
+        plot_name = plot_names[0]
+
+    predictions['geometry'] = predictions.apply(
+        lambda x: shapely.geometry.box(x.xmin, x.ymin, x.xmax, x.ymax), axis=1)
+    predictions = gpd.GeoDataFrame(predictions, geometry='geometry')
+
+    ground_df['geometry'] = ground_df.apply(lambda x: shapely.geometry.Point(x.x, x.y),
+                                            axis=1)
+    ground_df = gpd.GeoDataFrame(ground_df, geometry='geometry')
+
+    # Which points in boxes
+    result = gpd.sjoin(ground_df, predictions, op='within', how="left")
+    result = result.rename(
+        columns={
+            "label_left": "true_label",
+            "label_right": "predicted_label",
+            "image_path_left": "image_path"
+        })
+    result = result.drop(columns=["index_right"])
+
+    if savedir:
+        if root_dir is None:
+            raise AttributeError("savedir is {}, but root dir is None".format(savedir))
+        image = np.array(Image.open("{}/{}".format(root_dir, plot_name)))[:, :, ::-1]
+        image = visualize.plot_predictions(image, df=predictions)
+        image = visualize.plot_points(image, df=ground_df, color=(0, 165, 255))
+        cv2.imwrite("{}/{}".format(savedir, plot_name), image)
+
+    return result
+
+
+def point_recall(predictions, ground_df, root_dir=None, savedir=None):
+    """Evaluate the proportion on ground truth points overlap with predictions
+    submission can be submitted as a .shp, existing pandas dataframe or .csv path
+    For bounding box recall, see evaluate(). 
+    Args:
+        predictions: a pandas dataframe, if supplied a root dir is needed to give the relative path of files in df.name. The labels in ground truth and predictions must match. If one is numeric, the other must be numeric.
+        ground_df: a pandas dataframe, if supplied a root dir is needed to give the relative path of files in df.name
+        root_dir: location of files in the dataframe 'name' column.
+    Returns:
+        results: a dataframe of match bounding boxes
+        box_recall: proportion of true positives of box position, regardless of class
```

Comment:
May want to clarify this language given that this is a little different from box evaluation